compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/EnumClassLowering.kt

```diff
@@ -153,7 +153,7 @@ private class EnumClassLowering(val context: JvmBackendContext) : ClassLoweringP
                     }
                 })
 
-                body = enumConstructor.body // will be transformed later
+                body = enumConstructor.body?.patchDeclarationParents(this)
```

Comment:
Is this really needed? If so I think there is another issue with lost parent

Reply:
I thought the comment referred to the patch parents phase which runs a bit later in the pipeline. There is nothing else in EnumClassLowering which would fix the declarations here.

Which other issue did you have in mind?

Reply:
I mean what the exact reason that there is still uninitialized `parent` by this time. The best solution might be fixing parent there.

Reply:
What is happening is that the original body of the constructor is copied to a new protected constructor with a slightly different signature. Later on, `EnumClassBodyTransformer` runs, which updates the parameter references and constructor calls. In principle, this pass could also update the parent fields. However, to do this we'd have to duplicate the implementation from `PatchDeclarationParentsVisitor`, since the visitor in question runs over the whole enum class instead of just over the constructor.

It is much more expedient to fix the declaration parents when we copy the constructor body.

I think the comment in the original source code refers to the `makePatchParentsPhase` which runs sometime after `EnumClassLowering`.

Reply:
Thank you for explanation